### PR TITLE
Switch to ClusterRole/RoleBindings to get ManagedClusters to pass authorizat…

### PR DESCRIPTION
**Pull request by asincu**
…ion in Voltron

## Description

Switch ClusterRoles/ClusterRoleBindings to ClusterRole/RoleBindings to access managed clusters in a multi-tenant setup. This is needed by components that monitor and access data from the managed clusters. 

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/manager.go [hunk 1]
```diff
@@ -970,6 +975,52 @@ func (c *managerComponent) managerAllowTigeraNetworkPolicy() *v3.NetworkPolicy {
 	}
 }
 
+func (c *managerComponent) multiTenantManagedClustersAccess() []client.Object {
+	var objects []client.Object
+	objects = append(objects, &rbacv1.Role{
```
Comment (caseydavenport) [inline pkg/render/manager.go:980]: In multi-tenant systems, there is already going to be a role in this namespace - I don't think we should have multiple roles per-tenant that allow managed cluster acccess.

Comment (asincu) [reply pkg/render/manager.go]: I am only adding a RoleBinding in the tenant namespace.

**Review by caseydavenport — APPROVED** [review]
One minor comment on a comment but otherwise LGTM

File: pkg/render/apiserver.go [hunk 2]
```diff
@@ -1878,3 +1884,24 @@ rules:
...
+	var objects []client.Object
+	objects = append(objects, &rbacv1.ClusterRole{
+		TypeMeta:   metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: MultiTenantManagedClustersAccessClusterRoleName},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"projectcalico.org"},
+				Resources: []string{"managedclusters"},
+				Verbs: []string{
+					// The Authentication Proxy in Voltron checks if Enterprise Components (using impersonation headers for
```
[the remark this answers is not in the document]
Comment (asincu) [reply pkg/render/apiserver.go:1898]: I added some clarifications.

**Pull request merged**